sphinx/application.py

```diff
@@ -1033,19 +1035,24 @@ def add_latex_package(self, packagename, options=None):
         self.registry.add_latex_package(packagename, options)
 
     def add_lexer(self, alias, lexer):
-        # type: (str, Any) -> None
+        # type: (str, Union[Lexer, Type[Lexer]]) -> None
         """Register a new lexer for source code.
 
-        Use *lexer*, which must be an instance of a Pygments lexer class, to
-        highlight code blocks with the given language *alias*.
+        Use *lexer* to highlight code blocks with the given language *alias*.
 
         .. versionadded:: 0.6
+        .. versionchanged:: 2.1
+           Take a lexer class as an argument.  An instance of lexers are
+           also supported for a while.
```

Comment:
nits suggestion:

- An instance of lexers are still supported before Sphinx-4.0.

Reply:
okay